lib/puppet/face/facts.rb

```diff
@@ -75,13 +75,16 @@
         facts.name = Puppet[:node_name_value]
       end
 
-      Puppet::Node::Facts.indirection.terminus_class = :rest
-      server = Puppet::Node::Facts::Rest.server
-      Puppet.notice(_("Uploading facts for '%{node}' to: '%{server}'") % {
+      client = Puppet.runtime['http']
+      session = client.create_session
+      puppet = session.route_to(:puppet)
+
+      Puppet.notice(_("Uploading facts for '%{node}' to '%{server}'") % {
                     node: Puppet[:node_name_value],
-                    server: server})
+                    server: puppet.url.hostname})
 
-      Puppet::Node::Facts.indirection.save(facts, nil, :environment => Puppet.lookup(:current_environment))
+      puppet.put_facts(Puppet[:node_name_value], facts: facts, environment: Puppet.lookup(:current_environment).name.to_s)
+      nil
```

Comment:
Why are we returning `nil` here?

Reply:
ah, faces automatically prints out whatever the last line returns. Previously it was relying on `indirection.save` returning nil (I think). Let me check/confirm and update the commit message.

Reply:
Yeah it returns `nil`. I updated the commit message so it's clear why I added that.